packages/typir/src/features/conversion.ts

```diff
@@ -12,11 +12,13 @@ import { toArray } from '../utils/utils.js';
 export type ConversionMode =
     'IMPLICIT' | // coercion
     'EXPLICIT' | // casting
-    'BOTH'; // TODO does this make sense? Does IMPLICIT => EXPLICIT hold?
+    'NONE' |     // no conversion possible at all
+    'SELF';      // a type is always self-convertible to itself
 
 export interface TypeConversion {
     markAsConvertible(from: Type | Type[], to: Type | Type[], mode: ConversionMode): void
-    isConvertibleTo(from: Type, to: Type, mode: ConversionMode): boolean;
+    getConversion(from: Type, to: Type): ConversionMode;
```

Comment:
To me, the file is a bit hard to understand overall. Would it be possible to add a few comments?

Reply:
Example: What is it concerned with in general? Why does getConversion only resolve on NONE or SELF while there are four ConversionModes?

Reply:
I just pushed some more comments for this feature.